File: zcgl/src/main/java/com/bjprd/zcgl/widget/MessageDialog.java
package com.bjprd.zcgl.widget;

import android.app.Activity;
import android.app.Dialog;
import android.view.View;
import android.view.Window;
import android.widget.TextView;

import com.bjprd.zcgl.R;


/**
 * Created by 王少岩 on 2016/5/31.
 */
public class MessageDialog extends Dialog {

    // 变量
    private Activity mContext;
    private OnDialogFinishListener mListener;

    private TextView tvTitle;
    private TextView tvLeftButton;
    private TextView tvRightButton;
    private TextView tvMessage;
    private View line;

    public MessageDialog(Activity context, String msg) {
        super(context);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.dialog_alert_app);
        mContext = context;
        setCanceledOnTouchOutside(false);
        setCancelable(false);
        initControls(msg);
    }

    public MessageDialog(Activity context, String msg, boolean isShowLeftBtn, OnDialogFinishListener listener) {
        super(context);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.dialog_alert_app);
        mContext = context;
        mListener = listener;
        setCanceledOnTouchOutside(false);
        setCancelable(false);
        initControls(msg);
        if(!isShowLeftBtn){
            hideLeftBtn();
        }
    }

    private void initControls(String msg){
        tvTitle = (TextView) findViewById(R.id.tv_dialog_title);
        tvLeftButton = (TextView) findViewById(R.id.tv_dialog_left_button);
        tvRightButton = (TextView) findViewById(R.id.tv_dialog_right_button);
        tvMessage = (TextView) findViewById(R.id.tv_dialog_msg);
        line = findViewById(R.id.line_h);
        tvTitle.setText(mContext.getString(R.string.app_name));
        tvMessage.setText(msg);
        tvLeftButton.setOnClickListener(mOnClickListener);
        tvRightButton.setOnClickListener(mOnClickListener);
    }

    public void hideLeftBtn(){
        line.setVisibility(View.GONE);
        tvLeftButton.setVisibility(View.GONE);
    }

    public void setTitle(String title){
        tvTitle.setVisibility(View.VISIBLE);
        tvTitle.setText(title);
    }

    public void hideTitle(){
        tvTitle.setVisibility(View.GONE);
    }

    public void setMsg(String msg){
        tvMessage.setText(msg);
    }

    public void setDialogFinishListener(OnDialogFinishListener listener){
        mListener = listener;
    }

    private View.OnClickListener mOnClickListener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            int i = v.getId();
            if (i == R.id.tv_dialog_left_button) {
                if (mListener != null) {
                    mListener.onCancel();
                }
                MessageDialog.this.dismiss();

            } else if (i == R.id.tv_dialog_right_button) {
                if (mListener != null) {
                    mListener.onFinish();
                }
                MessageDialog.this.dismiss();

            }
        }
    };

    public interface OnDialogFinishListener{
        void onFinish();
        void onCancel();
    }
}

File: Zcgl/src/main/java/com/bjprd/zcgl/widget/MessageDialog.kt
package com.bjprd.zcgl.widget

import android.app.Activity
import android.app.Dialog
import android.view.View
import android.view.Window
import android.widget.TextView

import com.bjprd.zcgl.R


/**
 * Created by 王少岩 on 2016/5/31.
 */
class MessageDialog : Dialog {

    // 变量
    private var mContext: Activity? = null
    private var mListener: OnDialogFinishListener? = null

    private var tvTitle: TextView? = null
    private var tvLeftButton: TextView? = null
    private var tvRightButton: TextView? = null
    private var tvMessage: TextView? = null
    private var line: View? = null

    constructor(context: Activity, msg: String) : super(context) {
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        setContentView(R.layout.dialog_alert_app)
        mContext = context
        setCanceledOnTouchOutside(false)
        setCancelable(false)
        initControls(msg)
    }

    constructor(context: Activity, msg: String, isShowLeftBtn: Boolean, listener: OnDialogFinishListener) : super(context) {
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        setContentView(R.layout.dialog_alert_app)
        mContext = context
        mListener = listener
        setCanceledOnTouchOutside(false)
        setCancelable(false)
        initControls(msg)
        if (!isShowLeftBtn) {
            hideLeftBtn()
        }
    }

    private fun initControls(msg: String) {
        tvTitle = findViewById(R.id.tv_dialog_title) as TextView
        tvLeftButton = findViewById(R.id.tv_dialog_left_button) as TextView
        tvRightButton = findViewById(R.id.tv_dialog_right_button) as TextView
        tvMessage = findViewById(R.id.tv_dialog_msg) as TextView
        line = findViewById(R.id.line_h)
        tvTitle!!.text = mContext!!.getString(R.string.app_name)
        tvMessage!!.text = msg
        tvLeftButton!!.setOnClickListener(mOnClickListener)
        tvRightButton!!.setOnClickListener(mOnClickListener)
    }

    fun hideLeftBtn() {
        line!!.visibility = View.GONE
        tvLeftButton!!.visibility = View.GONE
    }

    fun setTitle(title: String) {
        tvTitle!!.visibility = View.VISIBLE
        tvTitle!!.text = title
    }

    fun hideTitle() {
        tvTitle!!.visibility = View.GONE
    }

    fun setMsg(msg: String) {
        tvMessage!!.text = msg
    }

    fun setDialogFinishListener(listener: OnDialogFinishListener) {
        mListener = listener
    }

    private val mOnClickListener = View.OnClickListener { v ->
        val i = v.id
        if (i == R.id.tv_dialog_left_button) {
            if (mListener != null) {
                mListener!!.onCancel()
            }
            this@MessageDialog.dismiss()

        } else if (i == R.id.tv_dialog_right_button) {
            if (mListener != null) {
                mListener!!.onFinish()
            }
            this@MessageDialog.dismiss()

        }
    }

    interface OnDialogFinishListener {
        fun onFinish()
        fun onCancel()
    }
}
